pkg/controllers/management/secretmigrator/clusters.go

```diff
@@ -857,6 +857,7 @@ func (m *Migrator) createOrUpdateSecret(secretName, secretNamespace string, data
 // and creates a secret to hold the sanitized data. If an owner is passed, the owner is set as an owner reference on the secret.
 func (m *Migrator) createOrUpdateSecretForCredential(secretName, secretNamespace, secretValue string, annotations map[string]string, owner runtime.Object, kind, field string) (*corev1.Secret, error) {
 	if secretValue == "" {
```

Comment:
We should check if secretName is empty before referencing it and log an additional message if so.